File: database-jdbc/aliyun-dms/src/main/scalajvm/com/peknight/database/jdbc/aliyun/dms/AliyunDmsException.scala
package com.peknight.database.jdbc.aliyun.dms

import com.peknight.error.Error

import java.sql.SQLException

/**
 * DMS API 异常封装为 SQLException
 */
case class AliyunDmsException(override val message: String, errorCode: Option[String] = None,
                              override val cause: Option[Error] = None)
  extends SQLException(message, null, cause.orNull) with Error

File: database-jdbc/aliyun-dms/src/main/java/com/peknight/database/jdbc/aliyun/dms/AliyunDmsException.java
package com.peknight.database.jdbc.aliyun.dms;

import java.sql.SQLException;

/**
 * DMS API 异常封装为 SQLException
 */
public class AliyunDmsException extends SQLException {

    private final String dmsErrorCode;
    private final String dmsRequestId;

    public AliyunDmsException(String message) {
        super(message);
        this.dmsErrorCode = null;
        this.dmsRequestId = null;
    }

    public AliyunDmsException(String message, Throwable cause) {
        super(message, cause);
        this.dmsErrorCode = null;
        this.dmsRequestId = null;
    }

    public AliyunDmsException(String dmsErrorCode, String message, String requestId) {
        super(message);
        this.dmsErrorCode = dmsErrorCode;
        this.dmsRequestId = requestId;
    }

    public String getDmsErrorCode() {
        return dmsErrorCode;
    }

    public String getDmsRequestId() {
        return dmsRequestId;
    }
}
